### PR TITLE
Improve docs about locales for date handling expression functions

### LINKED ISSUE #42325: to_datetime expression function example is broken on non-english locale.
**Describe the bug**
One `to_datetime` expression function example is broken on non-"en" locale.

**How to Reproduce**

The second example for the `to_datetime` function is `to_datetime('June 29, 2019 @ 12:34','MMMM d, yyyy @ HH:mm')` → `2019-06-29T12:34`. "June" is the month name in English and thus the parsing will only work if the user's locale is "en"-something. On a German locale it fails.

- The documentation of the function should include a note about the locale being used by default.
- The affected example should get a note next to the example result.

**QGIS and OS versions**

QGIS-Version | 3.18.0-Zürich | QGIS-Codeversion | bdef9fb328
-- | -- | -- | --
Kompiliert gegen Qt | 5.12.8 | Laufendes Qt | 5.12.8
Kompiliert mit GDAL/OGR | 3.0.4 | Läuft mit GDAL/OGR | 3.0.4
Kompiliert mit GEOS | 3.8.0-CAPI-1.13.1 | Läuft mit GEOS | 3.8.0-CAPI-1.13.1
Kompiliert mit SQLite | 3.31.1 | Läuft mit SQLite | 3.31.1
Gegen PDAL kompiliert | 2.0.1 | Läuft mit PDAL | 2.0.1 (git-version: Release)
PostgreSQL-Client-Version | 12.6 (Ubuntu 12.6-0ubuntu0.20.04.1) | SpatiaLite-Version | 4.3.0a
QWT-Version | 6.1.4 | QScintilla2-Version | 2.11.2
Kompiliert mit PROJ | 6.3.1 | Läuft mit PROJ | Rel. 6.3.1, February 10th, 2020
BS-Version | Ubuntu 20.04.2 LTS

### PR DESCRIPTION
**Pull request by kannes**
## Description
The second example for the `to_datetime` function is `to_datetime('June 29, 2019 @ 12:34','MMMM d, yyyy @ HH:mm')` → `2019-06-29T12:34`. "June" is the month name in English and thus the parsing will only work if the user's locale is "en"-something. On a German locale it fails.

This PR addes some notes on the usage, what happens of parsing fails and mentions the default. Fixes #42325.

image
image

I am not sure if putting `@qgis_locale` as default parameter is ok, just thought it would be nice to use that opportunity to explicitly highlight what the default is.

### CODE REVIEW

File: resources/function_help/json/format_date [hunk 1]
```diff
@@ -6,11 +6,11 @@
...
- {"arg":"language","optional":true,"description":"language (lowercase, two- or three-letter, <a href='https://en.wikipedia.org/wiki/List_of_ISO_639-1_codes'>ISO 639 language code</a>) used to format the date into a custom string"}
+ {"arg":"language","optional":true,"default":"@qgis_locale","description":"language (lowercase, two- or three-letter, <a href='https://en.wikipedia.org/wiki/List_of_ISO_639-1_codes'>ISO 639 language code</a>) used to format the date into a custom string. By default the currently set-up locale is used."}
```
Comment (nyalldawson) [inline resources/function_help/json/format_date:9]:
```suggestion
 {"arg":"language","optional":true,"default":"@qgis_locale","description":"language (lowercase, two- or three-letter, <a href='https://en.wikipedia.org/wiki/List_of_ISO_639-1_codes'>ISO 639 language code</a>) used to format the date into a custom string. By default the current QGIS user locale is used."}
```

(elsewhere too!)

Comment (kannes) [reply resources/function_help/json/format_date:9]: Much nicer, I thought my wording was weird. Thanks I will fix that.

**Comment by nirvn** [comment]
A note on the @qgis_locale being the default. While it's correct to state that the default language is that of the OS (or that being customized to in QGIS' settings), it doesn't rely on that variable. So, say someone sees this help and thinks he/she can overwrite the default QGIS-wide variable content to something else in a print layout scope, it'd have no effect.

**Comment by DelazJ** [comment]
Good point @nirvn. Maybe should we then avoid `@qgis_locale` and just be more literal, e.g "default":"current locale"?

**Comment by kannes** [comment]
Thanks, that is a very important point. I would prefer not showing any "fake" default value then but add a sentence to the main description.

**Comment by kannes** [comment]
Update wording and removed the variable. Ready from my side. :)

**Pull request merged**